tdp/cli/commands/deploy.py

```diff
@@ -98,16 +106,53 @@ def deploy(
         check_services_cleanliness(cluster_variables)
 
         operation_runner = OperationRunner(dag, ansible_executor, cluster_variables)
+        if resume:
+            if sources or targets:
+                raise click.BadParameter(
+                    f"sources or targets resume can not be used resume"
+                )
+            resumed_deployment = get_deployment(session_class, resume)
+            if resumed_deployment.operations[-1].state == "Success":
+                raise click.BadParameter(
+                    f"Nothing to resume, deployment n°{resume} was sucessful"
+                )
+            else:
+                click.echo(f"Resuming deployment n°{resume}")
+            if resumed_deployment.targets is not None:
+                targets = resumed_deployment.targets
+            if resumed_deployment.sources is not None:
+                sources = resumed_deployment.sources
+            if resumed_deployment.filter_expression is not None:
+                filter = resumed_deployment.filter_expression
         if sources:
             click.echo(f"Deploying from {sources}")
         elif targets:
             click.echo(f"Deploying to {targets}")
         else:
             click.echo(f"Deploying TDP")
         try:
-            operation_iterator = operation_runner.run_nodes(
-                sources=sources, targets=targets, filter_expression=filter
-            )
+            if resume:
+                operation_plan_to_resume = OperationPlan.from_dag(
+                    dag, targets, sources, filter
+                )
+                original_operations = [
+                    operation.name for operation in operation_plan_to_resume.operations
+                ]
+                succeeded_operations = [
+                    operation.operation
+                    for operation in resumed_deployment.operations
+                    if operation.state == "Success"
+                ]
+                remaining_operations = list(
+                    set(original_operations) - set(succeeded_operations)
+                )
+                operation_iterator = operation_runner.run_operations(
```

Comment:
With this, the only thing we know about the deployment in database is that it's not from the Dag, but we do not know that it's a resume.